src/cdk/drag-drop/drop-list-ref.ts

```diff
@@ -820,7 +820,101 @@ export class DropListRef<T = any> {
    * @param y Position of the item along the Y axis.
    */
   _getSiblingContainerFromPosition(item: DragRef, x: number, y: number): DropListRef | undefined {
-    return this._siblings.find(sibling => sibling._canReceive(item, x, y));
+    // Possible targets include siblings and 'this'
+    let targets = [this, ...this._siblings];
+
+    // Only consider targets where the drag postition is within the client rect
+    // (this avoids calling enterPredicate on each possible target)
+    let matchingTargets = targets.filter(ref => {
+      return ref._clientRect && isInsideClientRect(ref._clientRect, x, y);
```

Comment:
I don't think that we can depend purely on the `ClientRect`, because it doesn't account for overlapping containers or ones that have been scrolled out of view. That's why we have the `elementFromPoint` call inside `_canReceive`.

Reply:
The code does actually not depend solely on `ClientRect`. There is a call to `_canRecieve` on line 849 that should handle situations were the drop container is scrolled out of view or when there are elements overlapping the target drop container.

There is, however, an implicit assumption that the `ClientRect` of an inner drop container will only overlap with siblings that are its own ancestors in the drop container hierarchy. Although this is likely to be the most common case, it may not always be true. Let me see if I can find an elegant way to do away with this assumption.